File: 2002/4/9-4.rb
#!/usr/bin/env ruby

STDIN.readlines.each do |line|
  count = 0
  x = 1
  line.chars do |c|
    if x >= 0
      count += 1
      if c == 'e'
        x += 1
      else
        x -= 1
      end
    end
  end
  puts (x >= 0 ? -1 : count)
end

File: 2002/4/9-4.py
#!/usr/bin/env python

import sys

for line in sys.stdin.readlines():
    count = 0
    x = 1
    for l in line:
        if x >= 0:
            count += 1
            if l == 'e':
                x += 1
            else:
                x -= 1
    if x >= 0:
        print -1
    else:
        print count
